### PR TITLE
Fyst 1516 add email validation step to prior year access flow

### PR DESCRIPTION
**Pull request by DrewProebstel**
## Link to pivotal/JIRA issue
-https://codeforamerica.atlassian.net/browse/FYST-1516
## Is PM acceptance required? (delete one)
- Yes - don't merge until JIRA issue is accepted!

**Reminder**: merge main into this branch and get green tests before merging to main
## What was done?
- adds the `State_File_Archive_Intake` `State_File_Archive_Access_Log` and `State_File_Archive_Request` models
- adds a link in the about page for the get your pdf flow
- adds feature flags for the new flow
- adds email address page
- adds verification code page
## How to test?
- Describe the testing approach taken to verify the changes, including:
  - Hop on demo and test out the flow
- Specify any relevant testing environments used (e.g., development, staging, demo, Heroku).
- Risk Assessment
  - Risks or side effects associated with the changes and how they were mitigated.
  - Highlight areas that may need extra attention during code review or testing.
  - Paste SQL queries or output where relevant
## Screenshots (for visual changes)
Screenshot 2025-01-10 at 4 28 40 PM
Screenshot 2025-01-10 at 4 28 51 PM
Screenshot 2025-01-10 at 4 29 33 PM

TODO: 
Make something more beautiful than this in `app/jobs/archived_intake_email_verification_code_job.rb`

```
def priority
    PRIORITY_HIGH - 1
end
```

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -647,6 +647,33 @@
     t.index ["name"], name: "index_coalitions_on_name", unique: true
   end
 
+  create_table "completed_2023_tax_return_logs", force: :cascade do |t|
```
Comment (anisharamnani) [inline db/schema.rb:650]: ⚠️ these schema change shouldn't be here.

File: app/controllers/state_file/archived_intakes/verification_code_controller.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      def edit
+        @form = VerificationCodeForm.new
+        @email_address = params[:email_address]
+      end
+
+      def update
+        @form = VerificationCodeForm.new(verification_code_form_params)
+
+        if @form.save
+          redirect_to root_path
```
Comment (anisharamnani) [inline app/controllers/state_file/archived_intakes/verification_code_controller.rb:13]: I _think_ this is where I would put my new route?

File: app/controllers/state_file/archived_intakes/email_address_controller.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        end
+      end
+
+      private
+
+      def email_address_form_params
+        params.require(:state_file_archived_intakes_email_address_form).permit(:email_address)
+      end
+
+      def check_feature_flag
```
Comment (squanto) [inline app/controllers/state_file/archived_intakes/email_address_controller.rb:30]: < nitpick >
this before action looks like a good candidate for moving to `ArchivedIntakeController`

File: db/migrate/20250108231212_validate_foreign_keyfor_state_file_archived_intake_access_logs.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class ValidateForeignKeyforStateFileArchivedIntakeAccessLogs < ActiveRecord::Migration[7.1]
+  def change
+    validate_foreign_key :state_file_archived_intake_access_logs, :state_file_archived_intake_requests
```
Comment (arinchoi03) [inline db/migrate/20250108231212_validate_foreign_keyfor_state_file_archived_intake_access_logs.rb:3]: can you explain why we have two migrations, with one saying `validate: false` for this column but then this one validates?

File: app/controllers/state_file/archived_intakes/email_address_controller.rb [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+      end
+
+      def update
+        @form = EmailAddressForm.new(email_address_form_params)
+
+        if @form.save
+          archived_intake = StateFileArchivedIntake.find_by(email_address: @form.email_address)
+          session[:email_address] = @form.email_address
+          StateFileArchivedIntakeRequest.find_or_create_by(email_address: @form.email_address, ip_address: ip_for_irs, state_file_archived_intakes_id: archived_intake&.id )
+          create_state_file_access_log(0)
```
Comment (jenny-heath) [inline app/controllers/state_file/archived_intakes/email_address_controller.rb:16]: the nice thing about rails enums is you don't have to refer to them by their integer values, you can use the english word!

File: app/models/state_file_archived_intake_request.rb [hunk 6]
```diff
@@ -0,0 +1,38 @@
+# == Schema Information
+#
+# Table name: state_file_archived_intake_requests
+#
+#  id                             :bigint           not null, primary key
+#  details                        :jsonb
```
Comment (arinchoi03) [inline app/models/state_file_archived_intake_request.rb:6]: what goes into the `details`?

File: app/controllers/state_file/archived_intakes/verification_code_controller.rb [hunk 7]
```diff
@@ -0,0 +1,50 @@
+module StateFile
+  module ArchivedIntakes
+    class VerificationCodeController < ArchivedIntakeController
+      before_action :check_feature_flag
+      def edit
+        if current_request.access_locked?
+          redirect_to root_path
```
[the remark this answers is not in the document]
Comment (arinchoi03) [reply app/controllers/state_file/archived_intakes/verification_code_controller.rb:7]: story coming for: "you've been locked" page -- drew to add a note

File: app/jobs/archived_intake_email_verification_code_job.rb [hunk 8]
```diff
@@ -0,0 +1,12 @@
+class ArchivedIntakeEmailVerificationCodeJob < ApplicationJob
+  retry_on Mailgun::CommunicationError
+
+  def priority
+    PRIORITY_HIGH - 1 # Subtracting one to push to the top of the queue
+  end
+
+  # def perform(email_address: nil, locale:)
```
Comment (arinchoi03) [inline app/jobs/archived_intake_email_verification_code_job.rb:8]: remove this commented out stuff!

File: spec/controllers/state_file/archived_intake/archived_intake_controller_spec.rb [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+      allow(controller).to receive(:current_request).and_return(request_instance)
+    end
+
+    it 'creates a StateFileArchivedIntakeAccessLog with the correct attributes' do
+      expect(StateFileArchivedIntakeAccessLog).to receive(:create!).with(
+        event_type: event_type,
+        state_file_archived_intake_request: request_instance
+      ).and_return(access_log_instance)
+
+      controller.create_state_file_access_log(event_type)
```
Comment (arinchoi03) [inline spec/controllers/state_file/archived_intake/archived_intake_controller_spec.rb:47]: why does it create a access log here? is there an expectation missing?

**Review by arinchoi03 — COMMENTED** [review]
some comments that we discussed!

File: spec/factories/state_file/state_file_archived_intake_requests.rb [hunk 10]
```diff
@@ -0,0 +1,11 @@
+FactoryBot.define do
+  factory :state_file_archived_intake_request do
+    email_address { "geddy_lee@gmail.com" }
+    failed_attempts { 0 }
+    locked_at { nil }
+
+    trait :locked do
```
Comment (anisharamnani) [inline spec/factories/state_file/state_file_archived_intake_requests.rb:7]: 🤔 does it make sense to put this in the past? or does it have to be the current time, considering that sometimes when tests run on CI, current time can be an issue.

File: app/services/archived_intake_email_verification_code_service.rb [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+    VerificationCodeMailer.archived_intake_verification_code(
+      to: @email_address,
+      verification_code: verification_code,
+      locale: @locale
+    ).deliver_now
+  end
+
+  def self.request_code(**args)
+    new(**args).request_code
+  end
```
[the remark this answers is not in the document]
Comment (arinchoi03) [reply app/services/archived_intake_email_verification_code_service.rb:18]: looks like pattern introduced in https://github.com/codeforamerica/vita-min/pull/1245, no explanation given but looking at it now, wondering if this is just a shortcut to easily call
`ArchivedIntakeEmailVerificationCodeService.request_code(**args)` instead of doing `ArchivedIntakeEmailVerificationCodeService.new(email_address:,locale: ).request_code`

File: db/migrate/20250108231212_validate_foreign_keyfor_state_file_archived_intake_access_logs.rb (hunk 4 above)
Comment (anisharamnani) [reply db/migrate/20250108231212_validate_foreign_keyfor_state_file_archived_intake_access_logs.rb:3]: this was a suggestion from the [strong_migrations ](https://github.com/ankane/strong_migrations) gem. i've attached a screenshot of the general rule. the idea is that adding the foreign key & validating the migration lock both tables. 

granted, there is no data in these tables, so locking them shouldn't be an issue, but strong migrations was loud enough that the only work around was to create two migrations.
Screenshot 2025-01-13 at 3 53 00 PM

File: spec/controllers/state_file/archived_intake/archived_intake_controller_spec.rb [hunk 12]
```diff
@@ -0,0 +1,82 @@
+require 'rails_helper'
+
+describe StateFile::ArchivedIntakes::ArchivedIntakeController, type: :controller do
+  let(:ip_address) { '192.168.0.1' }
+  let(:email_address) { 'test@example.com' }
+  let(:request_instance) { instance_double(StateFileArchivedIntakeRequest) }
+
+  before do
+    allow(controller).to receive(:ip_for_irs).and_return(ip_address)
+    allow(session).to receive(:[]).with(:email_address).and_return(email_address)
```
Comment (jenny-heath) [inline spec/controllers/state_file/archived_intake/archived_intake_controller_spec.rb:10]: really minor but you can actually just do this:
```suggestion
    session[:email_address] = email_address
```

File: spec/controllers/state_file/archived_intake/archived_intake_controller_spec.rb [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+  end
+
+  describe '#current_request' do
+    it 'finds the StateFileArchivedIntakeRequest by IP and email address' do
+      expect(StateFileArchivedIntakeRequest).to receive(:find_by).with(
+        ip_address: ip_address,
+        email_address: email_address
+      ).and_return(request_instance)
+
+      expect(controller.current_request).to eq(request_instance)
```
Comment (jenny-heath) [inline spec/controllers/state_file/archived_intake/archived_intake_controller_spec.rb:20]: i think it seems unnecessary to expect the `find_by`. it would probably be sufficient to test that the return value of `current_request` is the right record.
```suggestion
      expect(controller.current_request).to eq(request_instance)
```

File: spec/controllers/state_file/archived_intake/archived_intake_controller_spec.rb [hunk 14]
```diff
@@ -0,0 +1,82 @@
+require 'rails_helper'
+
+describe StateFile::ArchivedIntakes::ArchivedIntakeController, type: :controller do
+  let(:ip_address) { '192.168.0.1' }
+  let(:email_address) { 'test@example.com' }
+  let(:request_instance) { instance_double(StateFileArchivedIntakeRequest) }
```
Comment (jenny-heath) [inline spec/controllers/state_file/archived_intake/archived_intake_controller_spec.rb:6]: unless you're mocking methods on this model instance (which i don't see anywhere), i think it would make more sense to just create an actual `StateFileArchivedIntakeRequest` from the factory
```suggestion
  let(:request_instance) { create :state_file_archived_intake_request, ip_address: ip_address }
```

File: spec/controllers/state_file/archived_intake/archived_intake_controller_spec.rb [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+      expect(controller.current_request).to eq(request_instance)
+    end
+
+    it 'returns nil if no request is found' do
+      expect(StateFileArchivedIntakeRequest).to receive(:find_by).with(
+        ip_address: ip_address,
+        email_address: email_address
+      ).and_return(nil)
+
+      expect(controller.current_request).to be_nil
```
Comment (jenny-heath) [inline spec/controllers/state_file/archived_intake/archived_intake_controller_spec.rb:29]: same as above
```suggestion
      expect(controller.current_request).to be_nil
```

File: spec/factories/state_file/state_file_archived_intake_requests.rb
Comment (jenny-heath) [inline spec/factories/state_file/state_file_archived_intake_requests.rb:1]: do you need to run `annotate`? this file doesn't appear to have annotations like the rest of the factories

File: spec/controllers/state_file/archived_intake/archived_intake_controller_spec.rb [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+    end
+  end
+
+  describe '#create_state_file_access_log' do
+    let(:event_type) { 'access_granted' }
+    let(:access_log_instance) { instance_double(StateFileArchivedIntakeAccessLog) }
```
Comment (jenny-heath) [inline spec/controllers/state_file/archived_intake/archived_intake_controller_spec.rb:35]: same as above
```suggestion
    let(:access_log_instance) { create : state_file_archived_intake_access_log }
```

File: spec/controllers/state_file/archived_intake/archived_intake_controller_spec.rb [hunk 17]
```diff
@@ -0,0 +1,82 @@
...
+      allow(controller).to receive(:current_request).and_return(request_instance)
+    end
+
+    it 'creates a StateFileArchivedIntakeAccessLog with the correct attributes' do
+      expect(StateFileArchivedIntakeAccessLog).to receive(:create!).with(
+        event_type: event_type,
+        state_file_archived_intake_request: request_instance
+      ).and_return(access_log_instance)
+
+      controller.create_state_file_access_log(event_type)
```
Comment (jenny-heath) [inline spec/controllers/state_file/archived_intake/archived_intake_controller_spec.rb:47]: i think something like this would be more consistent with our current testing style. theres something that feels odd to me about stubbing all the methods instead of just testing the output
```suggestion
      result = controller.create_state_file_access_log(event_type)
      
      expect(result).to be_a(StateFileArchivedIntakeAccessLog)
      expect(result.event_type).to eq event_type
      expect(result.state_file_archived_intake_request).to eq request_instance      
```

File: spec/controllers/state_file/archived_intake/archived_intake_controller_spec.rb [hunk 18]
```diff
@@ -0,0 +1,82 @@
...
+    end
+
+    it 'raises an error if the log cannot be created' do
+      allow(StateFileArchivedIntakeAccessLog).to receive(:create!).and_raise(ActiveRecord::RecordInvalid)
+
+      expect {
+        controller.create_state_file_access_log(event_type)
+      }.to raise_error(ActiveRecord::RecordInvalid)
+    end
```
Comment (jenny-heath) [inline spec/controllers/state_file/archived_intake/archived_intake_controller_spec.rb:56]: correct me if i'm wrong but does this just test that an error thrown by a method within a method will bubble up to the outer method...?

File: app/models/state_file_archived_intake_request.rb (hunk 6 above)
Comment (anisharamnani) [reply app/models/state_file_archived_intake_request.rb:6]: 🤔 this might have been a copy pasta mistake on my part. i think we can remove it.

File: spec/controllers/state_file/archived_intake/verification_code_controller_spec.rb [hunk 19]
```diff
@@ -0,0 +1,101 @@
+require "rails_helper"
+
+RSpec.describe StateFile::ArchivedIntakes::VerificationCodeController, type: :controller do
+  let(:current_request) { create(:state_file_archived_intake_request, failed_attempts: 0) }
```
Comment (jenny-heath) [inline spec/controllers/state_file/archived_intake/verification_code_controller_spec.rb:4]:
```suggestion
  let(:current_request) { create(:state_file_archived_intake_request, email_address: email_address, failed_attempts: 0) }
```

File: spec/controllers/state_file/archived_intake/verification_code_controller_spec.rb [hunk 20]
```diff
@@ -0,0 +1,101 @@
...
+  let(:email_address) { "test@example.com" }
+  let(:valid_verification_code) { "123456" }
+  let(:invalid_verification_code) { "654321" }
+
+  before do
+    Flipper.enable(:get_your_pdf)
+    allow(controller).to receive(:current_request).and_return(current_request)
+    allow(current_request).to receive(:email_address).and_return(email_address)
```
Comment (jenny-heath) [inline spec/controllers/state_file/archived_intake/verification_code_controller_spec.rb:12]:
```suggestion
```

File: spec/controllers/state_file/archived_intake/verification_code_controller_spec.rb [hunk 21]
```diff
@@ -0,0 +1,101 @@
...
+        allow(current_request).to receive(:access_locked?).and_return(false)
+      end
+
+      it "renders the edit template with a new VerificationCodeForm and queues a job" do
+        expect(ArchivedIntakeEmailVerificationCodeJob).to receive(:perform_later).with(
+          email_address: email_address,
+          locale: :en
+        )
+
+        get :edit
```
Comment (jenny-heath) [inline spec/controllers/state_file/archived_intake/verification_code_controller_spec.rb:40]: you can also do this
```suggestion
        expect {
          get :edit
        }.to have_enqueued_job(ArchivedIntakeEmailVerificationCodeJob).with(
          email_address: email_address,
          locale: :en
        )
```

File: spec/controllers/state_file/archived_intake/verification_code_controller_spec.rb [hunk 22]
```diff
@@ -0,0 +1,101 @@
...
+
+      it "locks the account and redirects to root path after multiple failed attempts" do
+        current_request.update!(failed_attempts: 1)
+
+        expect {
+          post :update, params: { state_file_archived_intakes_verification_code_form: { verification_code: invalid_verification_code } }
+        }.to change(StateFileArchivedIntakeAccessLog, :count).by(2)
+
+        log = StateFileArchivedIntakeAccessLog.last
+        expect(log.event_type).to eq("client_lockout_begin")
```
Comment (jenny-heath) [inline spec/controllers/state_file/archived_intake/verification_code_controller_spec.rb:93]: might make sense to also check the other access log

File: app/forms/state_file/archived_intakes/email_address_form.rb [hunk 23]
```diff
@@ -0,0 +1,15 @@
+module StateFile
+  module ArchivedIntakes
+    class EmailAddressForm < Form
+      attr_accessor :email_address
+
+      validates :email_address, 'valid_email_2/email': true
+      validates :email_address, presence: true
```
Comment (jenny-heath) [inline app/forms/state_file/archived_intakes/email_address_form.rb:7]:
```suggestion
      validates :email_address, presence: true, 'valid_email_2/email': true
```

File: app/forms/state_file/archived_intakes/email_address_form.rb [hunk 24]
```diff
@@ -0,0 +1,15 @@
...
+
+      def initialize(attributes = {})
+        super
+        assign_attributes(attributes)
+      end
```
Comment (jenny-heath) [inline app/forms/state_file/archived_intakes/email_address_form.rb:12]: i think `super` already does `assign_attributes(attributes)`
```suggestion
```

File: app/forms/state_file/archived_intakes/verification_code_form.rb [hunk 25]
```diff
@@ -0,0 +1,24 @@
+module StateFile
+  module ArchivedIntakes
+    class VerificationCodeForm < Form
+      attr_accessor :verification_code, :email_address
+
+      validates :verification_code, presence: true
+      def initialize(attributes = {}, email_address: nil)
+        super(attributes)
+        @email_address = email_address
+        assign_attributes(attributes)
```
Comment (jenny-heath) [inline app/forms/state_file/archived_intakes/verification_code_form.rb:10]: same as above
```suggestion
```

File: app/forms/state_file/archived_intakes/verification_code_form.rb [hunk 26]
```diff
@@ -0,0 +1,24 @@
+module StateFile
+  module ArchivedIntakes
+    class VerificationCodeForm < Form
+      attr_accessor :verification_code, :email_address
```
Comment (jenny-heath) [inline app/forms/state_file/archived_intakes/verification_code_form.rb:4]: sorry i should have caught this during the mob review but why `attr_accessor` instead of `set_attributes_for`? is it because we're not actually updating any db records?

File: app/jobs/archived_intake_email_verification_code_job.rb [hunk 27]
```diff
@@ -0,0 +1,11 @@
+class ArchivedIntakeEmailVerificationCodeJob < ApplicationJob
+  retry_on Mailgun::CommunicationError
+
+  def priority
+    PRIORITY_HIGH - 1 # Subtracting one to push to the top of the queue
```
Comment (jenny-heath) [inline app/jobs/archived_intake_email_verification_code_job.rb:5]: is this normal? seems weird to me

File: app/jobs/archived_intake_email_verification_code_job.rb
Comment (jenny-heath) [inline app/jobs/archived_intake_email_verification_code_job.rb:1]: pretty minor but could we have a test for this?

File: app/models/state_file_archived_intake_request.rb [hunk 28]
```diff
@@ -0,0 +1,37 @@
...
+  def self.maximum_attempts
+    2
+  end
+
+  def increment_failed_attempts
+    super
+    if attempts_exceeded? && !access_locked?
+      lock_access!
+    end
+  end
```
Comment (jenny-heath) [inline app/models/state_file_archived_intake_request.rb:36]: this could be tested i think

File: app/models/state_file_archived_intake_request.rb [hunk 29]
```diff
@@ -0,0 +1,37 @@
...
+# Indexes
+#
+#  idx_on_state_file_archived_intakes_id_31501c23f8  (state_file_archived_intakes_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (state_file_archived_intakes_id => state_file_archived_intakes.id)
+#
+class StateFileArchivedIntakeRequest < ApplicationRecord
+  # Include default devise modules. Others available are:
```
Comment (jenny-heath) [inline app/models/state_file_archived_intake_request.rb:23]: did you mean to leave this in? if so i don't understand it

File: app/services/archived_intake_email_verification_code_service.rb
Comment (jenny-heath) [inline app/services/archived_intake_email_verification_code_service.rb:1]: this also doesn't appear to have a test? maybe i'm missing something

File: app/views/state_file/state_file_pages/about_page.html.erb [hunk 30]
```diff
@@ -42,4 +42,13 @@
     </div>
   <% end %>
 </section>
-<% end %>
\ No newline at end of file
+<% end %>
+
+<div>
```
Comment (jenny-heath) [inline app/views/state_file/state_file_pages/about_page.html.erb:47]: not like it really matters but this div will be present but empty when the feature is not enabled. i'm not sure how long we intend to keep the feature flag but if it's going away when the feature launches (as in, we delete `if Flipper.enabled?...` etc) i wouldn't bother changing this

File: db/migrate/20250108221853_state_file_archived_intake_requests.rb [hunk 31]
```diff
@@ -0,0 +1,14 @@
...
+    create_table :state_file_archived_intake_requests do |t|
+      t.belongs_to :state_file_archived_intakes, foreign_key: true
+      t.string 'ip_address'
+      t.string 'email_address'
+      t.jsonb 'details', default: '{}'
+      t.timestamps
+    end
+
+    remove_foreign_key :state_file_archived_intake_access_logs, :state_file_archived_intakes
+    safety_assured { remove_column :state_file_archived_intake_access_logs, :state_file_archived_intakes_id }
```
Comment (jenny-heath) [inline db/migrate/20250108221853_state_file_archived_intake_requests.rb:12]: do these lines do different things? they look redundant to me but i'm by no means an expert on this kind of thing

File: db/migrate/20250113222716_remove_state_file_archived_intake_requests_fields.rb [hunk 32]
```diff
@@ -0,0 +1,8 @@
+class RemoveStateFileArchivedIntakeRequestsFields < ActiveRecord::Migration[7.1]
+  def change
+    safety_assured do
+      remove_column :state_file_archived_intake_access_logs, :ip_address
+      remove_column :state_file_archived_intake_requests, :details
```
Comment (jenny-heath) [inline db/migrate/20250113222716_remove_state_file_archived_intake_requests_fields.rb:5]: this is silly but since the migration that adds this column is also on this PR, couldn't you just not add it?

File: spec/factories/state_file/state_file_archived_intake_requests.rb
Comment (jenny-heath) [reply spec/factories/state_file/state_file_archived_intake_requests.rb:1]: ohh i think it's because this file is randomly in the `state_file` directory. could we move it up a level?

File: spec/features/state_file/prior_year_access_spec.rb [hunk 33]
```diff
@@ -0,0 +1,24 @@
...
+require 'axe-capybara'
+require 'axe-rspec'
+
+RSpec.feature "accessing a prior year PDF" , js: true, active_job: true do
+  before do
+    allow_any_instance_of(Routes::StateFileDomain).to receive(:matches?).and_return(true)
+    Flipper.enable(:get_your_pdf)
+  end
+
+  it "has content" do
```
Comment (jenny-heath) [inline spec/features/state_file/prior_year_access_spec.rb:11]: this could say a little more, for example `it "allows a user to enter an email address and offboards them when they enter an incorrect code"` (if that's what's happening here?)

File: app/jobs/archived_intake_email_verification_code_job.rb (hunk 27 above)
Comment (arinchoi03) [reply app/jobs/archived_intake_email_verification_code_job.rb:5]: this seemed weird to me too but it seems like we use this pattern 🙈 maybe there should be `PRIORITY_IMMEDIATE` or soemthing

File: db/schema.rb [hunk 34]
```diff
@@ -10,7 +10,9 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema[7.1].define(version: 2025_01_08_233940) do
+ActiveRecord::Schema[7.1].define(version: 2025_01_13_222716) do
+  create_schema "analytics"
```
Comment (arinchoi03) [inline db/schema.rb:14]: is this something we want to add?

File: spec/features/state_file/prior_year_access_spec.rb [hunk 35]
```diff
@@ -0,0 +1,24 @@
+require "rails_helper"
+require 'axe-capybara'
+require 'axe-rspec'
```
Comment (jenny-heath) [inline spec/features/state_file/prior_year_access_spec.rb:3]: what are these needed for?

File: app/jobs/archived_intake_email_verification_code_job.rb [hunk 36]
```diff
@@ -0,0 +1,11 @@
...
+  end
+
+  def perform(email_address:, locale:)
+    ArchivedIntakeEmailVerificationCodeService.request_code(email_address: email_address, locale: locale)
```
Comment (anisharamnani) [inline app/jobs/archived_intake_email_verification_code_job.rb:9]: [pebble] also i wanted to point out here, if you're renaming this file in your PR, you'd want to delete the old one too. i don't see that happening in the PR.

File: app/views/state_file/state_file_pages/about_page.html.erb [hunk 37]
```diff
@@ -42,4 +42,13 @@
...
+<% end %>
+
+<% if Flipper.enabled?(:get_your_pdf) %>
+<div>
+    <%= t(".looking_for_return_html")%>
+      <div class = "spacing-above-10">
+        <%= link_to t(".tax_return_link"), state_file_archived_intakes_edit_email_address_path %>
+      </div>
+</div>
+<% end %>
```
Comment (arinchoi03) [inline app/views/state_file/state_file_pages/about_page.html.erb:54]: [nitpick] can we fix the indent?

File: config/routes.rb [hunk 38]
```diff
@@ -1,4 +1,5 @@
 Rails.application.routes.draw do
+  devise_for :state_file_archived_intake_requests
```
Comment (arinchoi03) [inline config/routes.rb:2]: [dust] why is this at the top of the file, can it be closer to the routes that are relevant?

File: spec/controllers/state_file/archived_intake/archived_intake_controller_spec.rb [hunk 39]
```diff
@@ -0,0 +1,63 @@
...
+      allow(controller).to receive(:current_request).and_return(request_instance)
+    end
+
+    it 'creates a StateFileArchivedIntakeAccessLog with the correct attributes' do
+      result = controller.create_state_file_access_log(event_type)
+
+      expect(result).to be_a(StateFileArchivedIntakeAccessLog)
+      expect(result.event_type).to eq event_type
+      expect(result.state_file_archived_intake_request).to eq request_instance
+    end
```
Comment (arinchoi03) [inline spec/controllers/state_file/archived_intake/archived_intake_controller_spec.rb:37]: [pebble] can we add an expectation testing what would happen if `current_request` is `nil` since that's a potential outcome?

**Review by arinchoi03 — APPROVED** [review]
lgtm! just a small nitpicks/pebble

**Review by jenny-heath — APPROVED** [review]
great job!
